src/scenic/syntax/veneer.py

```diff
@@ -111,1000 +132,1341 @@
-	"""The :grammar:`angle to <vector>` operator (using the position of ego as the reference)."""
-	X = toVector(X, '"angle to X" with X not a vector')
-	return ego().angleTo(X)
-
-def AngleFrom(X, Y):
-	"""The :grammar:`angle from <vector> to <vector>` operator."""
-	X = toVector(X, '"angle from X to Y" with X not a vector')
-	Y = toVector(Y, '"angle from X to Y" with Y not a vector')
-	return X.angleTo(Y)
+    """The :grammar:`angle to <vector>` operator (using the position of ego as the reference)."""
+    X = toVector(X, '"angle to X" with X not a vector')
+    return ego().angleTo(X)
+
+def AngleFrom(X=None, Y=None):
+    """The :grammar:`angle from <vector> to <vector>` operator."""
+    assert X is not None or Y is not None
+    if X is None:
+        X = ego()
+    X = toVector(X, '"angle from X to Y" with X not a vector')
+    if Y is None:
+        Y = ego()
+    Y = toVector(Y, '"angle from X to Y" with Y not a vector')
+    return X.angleTo(Y)
+
+def AltitudeTo(X):
+    """The :grammar:`angle to <vector>` operator (using the position of ego as the reference)."""
+    X = toVector(X, '"altitude to X" with X not a vector')
+    return ego().altitudeTo(X)
+
+def AltitudeFrom(X=None, Y=None):
+    """The :grammar:`altitude from <vector> to <vector>` operator."""
+    assert X is not None or Y is not None
+    if X is None:
+        X = ego()
+    X = toVector(X, '"altitude from X to Y" with X not a vector')
+    if Y is None:
+        Y = ego()
+    Y = toVector(Y, '"altitude from X to Y" with Y not a vector')
+    return X.altitudeTo(Y)
 
 def Follow(F, X, D):
-	"""The :grammar:`follow <field> from <vector> for <number>` operator."""
-	if not isinstance(F, VectorField):
-		raise RuntimeParseError('"follow F from X for D" with F not a vector field')
-	X = toVector(X, '"follow F from X for D" with X not a vector')
-	D = toScalar(D, '"follow F from X for D" with D not a number')
-	pos = F.followFrom(X, D)
-	heading = F[pos]
-	return OrientedPoint(position=pos, heading=heading)
+    """The :grammar:`follow <field> from <vector> for <number>` operator."""
+    if not isA(F, VectorField):
+        raise TypeError('"follow F from X for D" with F not a vector field')
+    X = toVector(X, '"follow F from X for D" with X not a vector')
+    D = toScalar(D, '"follow F from X for D" with D not a number')
+    pos = F.followFrom(X, D)
+    orientation = F[pos]
+    return OrientedPoint._with(position=pos, parentOrientation=orientation)
+
+def VisibleFromOp(region, base):
+    """The :grammar:`<region> visible from <point>` operator."""
+    region = toType(region, Region, '"X visible from Y" with X not a Region')
+    if not isA(base, Point):
+        raise TypeError('"X visible from Y" with Y not a Point')
+    return region.intersect(base.visibleRegion)
 
 def CanSee(X, Y):
-	"""The :scenic:`{X} can see {Y}` polymorphic operator.
+    """The :scenic:`{X} can see {Y}` polymorphic operator.
+
+    Allowed forms::
+
+        <point> can see <point>
```

Comment:
The  case where `Y` is a vector should not be dropped: it makes sense and is documented. It looks like we have no tests where `Y` is a pair or triple of coordinates, but I suspect it would now fail since we're not using `toVector`.